onmt/translate/beam_search.py

```diff
@@ -93,8 +93,8 @@ def advance(self, log_probs, attn):
                 if fail:
                     log_probs[bk] = -10e20
 
-        alpha = self.global_scorer.alpha
-        length_penalty = ((5.0 + (self.alive_seq.shape[1])) / 6.0) ** alpha
+        length_penalty = self.global_scorer.length_penalty(
+            step, alpha=self.global_scorer.alpha)
```

Comment:
Should be `step + 1`.

Reply:
Really? ``step`` already starts at 1 since ``alive_seq`` has the bos token.

Reply:
Sorry, I assumed it was 0-indexed. But then I see another issue from the previous PR: `if step < self.min_length` should be `if step <= self.min_length`.

Reply:
Yeah. The difference is I guess whether the EOS and BOS count in that length. The tests verify that this matches how ``Beam`` does it. So, BOS counts towards the length and EOS can be the final token bringing the sequence to length ``min_length`` (I think - would have to look at the tests to make sure)

Reply:
`min_length` and `max_length` should refer to sequence length visible to the user in my opinion, so that excludes BOS and EOS.

Reply:
Sounds good. I updated that in both beam and beam_search. I expect it changes scores if you use the ``min_length`` parameter but I can't run the scores right now.

Reply:
Thanks!